components/script/dom/htmlimageelement.rs

```diff
@@ -1159,6 +1172,47 @@ impl HTMLImageElement {
         }
     }
 
+    // Step 2 for <https://html.spec.whatwg.org/multipage/#dom-img-decode>
+    fn react_to_decode_image_sync_steps(&self, promise: Rc<Promise>) {
+        let document = document_from_node(self);
+        // Step 2.1 of <https://html.spec.whatwg.org/multipage/#dom-img-decode>
+        if !document.is_fully_active() ||
+            matches!(self.current_request.borrow().state, State::Broken)
+        {
+            promise.reject_native(&DOMException::new(
+                &document.global(),
+                DOMErrorName::EncodingError,
+            ));
+        } else if matches!(
```

Comment:
Here I would add  a comment that this does not follow the spec, but it is discussed at https://github.com/whatwg/html/issues/4217